internal/v2/s2av2.go

```diff
@@ -0,0 +1,72 @@
+package s2av2
+
+import (
+	"errors"
+	"context"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tlsconfigstore"
+	"github.com/google/s2a-go/internal/handshaker"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	config   *tls.Config
+	isClient bool
+}
+
+func NewClientCreds() (*credentials.TransportCredentials) {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForClient(),
+		isClient: true,
+	}
+}
+
+func NewServerCreds() *credentials.TransportCredentials {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForServer(),
+		isClient: false,
+	}
+}
+
+func (c *s2av2TransportCreds) ClientHandShake(ctx context.Context, serverAuthority string, rawConn net.Conn) (_ net.Conn, _ credentials.AuthInfo, err error) {
```

Comment:
Since the original handshaker library offloaded the entire handshake to S2A, do we need to reimplement the handshaker interface to use it how it's used [here](https://github.com/google/s2a-go/blob/main/s2a.go#:~:text=secConn%2C%20authInfo%2C%20err%20%3A%3D%20chs.ClientHandshake(context.Background()))

Reply:
Nope, you won't have a handshaker interface for v2.

Reply:
I see, so instead of calling the handshaker interface methods like NewClientHandshaker and ClientHandshake, we make use of the libraries(tls_config_store, remote_signer and cert_verifier) which will make the RPCs to S2Av2, to just offload the certain operations in the handshake?

Reply:
Roughly yes. In a bit more detail, the flow should be:

```
func ClientHandshake(...) ... {
    // 1. Create a grpc.ClientConn instance to the S2A, using the API in [A]
    // 2. Build a stub from the grpc.ClientConn (similar to what you did in the echo examples when you called grpcpb.EchoClient)
   // 3. Get a *tls.Config from the TLS config store. You will probably need to pass the stub to the TLS Config store's APIs.
   // 4. return credentials.NewTls(your TLS config).ClientHandshake(...)
}
```

and similarly for `ServerHandshake`.

[A] https://github.com/google/s2a-go/blob/a941a3f6a219d2131d532a9abe27d7bd40244d35/internal/handshaker/service/service.go#L41

Reply:
Disclaimer: this is just how I pictured it in my head, please push back if you prefer a different route :)

Reply:
Thanks for the explanation Matt!